src/views/viewCommands.ts

```diff
@@ -447,12 +449,32 @@ export class ViewCommands implements Disposable {
         void (await Container.git.stageFile(node.repoPath, node.file.fileName));
     }
 
+    private async stageDirectory(node: ViewNode) {
+
+        // TODO: get fileNodes from node
+        const fileNodes: (CommitFileNode | StatusFileNode)[] = [];
+
+        for (const fileNode of fileNodes) {
+            void (await this.stageFile(fileNode));
```

Comment:
I'm not sure if staging each file individually is the way to do it or if there is a way to pass a folder to `Container.git.stageFile`.

Reply:
I think I would recommend adding a `stageDirectory` method to `GitService`, and it could call the same underlying `Git.add` method since the `fileName` parameter can really be a pathspec (and probably should be renamed)

See here: https://git-scm.com/docs/git-add#git-add-ltpathspecgt82308203

Reply:
And similar should be true for unstaging -- `Git.reset` can take a set of paths (although not a pathspec)

See here: https://git-scm.com/docs/git-reset#git-reset-emgitresetem-qlttree-ishgt--ltpathsgt82308203